### PR TITLE
MAPIStoreAppointmentWrapper: in _computeGlobalObjectIds create id for ev...

### PR DESCRIPTION
**Pull request by javierag**
...ent if it don't have none

This is to prevent a strlen crash when the event id is NULL.

### CODE REVIEW

**Comment by blaxter** [comment]
This seems to me like a workaround to avoid the real problem, if the event has no uid probably there is something really wrong with either the workflow or the data itself.

**Comment by javierag** [comment]
Yes, unfortunately the code path to reach this state this is unknown for me.

**Comment by blaxter** [comment]
Is this a crash report? which one? Do you have a way to reproduce it?

**Comment by jacalvo** [comment]
:+1: to blaxter comment, I don't think we should merge this if we don't know the original reason why that uid is unset...

File: OpenChange/MAPIStoreAppointmentWrapper.m [hunk 1]
```diff
@@ -1725,11 +1726,16 @@ - (void) _computeGlobalObjectIds
   NSData *binPrefix;
   TALLOC_CTX *localMemCtx;
 
-  localMemCtx = talloc_zero (NULL, TALLOC_CTX);
+  localMemCtx = talloc_zero(NULL, TALLOC_CTX);
```
Comment (sixstone-qq) [inline OpenChange/MAPIStoreAppointmentWrapper.m]: In Objective-C, the pure C procedures have spaces before parameters

**Comment by sixstone-qq** [comment]
@javierag Once we find the reason, we could merge it and log it to avoid crashing.

**Comment by javierag** [comment]
This was found in this crash https://tracker.zentyal.org/issues/3049
I have tried to reproduce it playing with attachments but not succes

**Comment by blaxter** [comment]
I'll try to reproduce this or at least understand the flow about this method, at first glance I've seen that this uid is generated once the event is being saved (and the nameInContainer) so afaik if we are hitting this crash probably is because we are working with an event not being save (either that or bad store/retrieve from database, which is less likely I think). Then we have to understand if that's possible.

Like I said before, I'll try to understand this once I had the time (maybe 2016? nah just joking... 2022 probably).

File: OpenChange/MAPIStoreAppointmentWrapper.m [hunk 2]
```diff
@@ -1725,11 +1726,16 @@ - (void) _computeGlobalObjectIds
...
 
   memset (&newGlobalId, 0, sizeof (struct GlobalObjectId));
 
   uid = [event uid];
```
Comment (sixstone-qq) [inline OpenChange/MAPIStoreAppointmentWrapper.m]: is uid a Obj-C object? Then use `if (!uid)`

Comment (jacalvo) [inline OpenChange/MAPIStoreAppointmentWrapper.m]: nil == NULL is true as well, anyway I also prefer if (!uid)

Comment (blaxter) [inline OpenChange/MAPIStoreAppointmentWrapper.m]: nil is `(id)0`
null is `(void *)0`

so `nil == NULL` is `(id)0 == (id)(void*)0`

A proper compiler probably will complain about `uid == NULL` mismatch types comparison. 

TL;DR: `if (!uid)`

Comment (sixstone-qq) [inline OpenChange/MAPIStoreAppointmentWrapper.m]: as @blaxter has pointed out, nil != NULL

**Comment by sixstone-qq** [comment]
This has been superseded by #112
